lib/fennec/evaluator/create_permission/request.ex

```diff
@@ -0,0 +1,72 @@
+defmodule Fennec.Evaluator.CreatePermission.Request do
+  @moduledoc false
+
+  import Fennec.Evaluator.Helper
+  alias Jerboa.Format.Body.Attribute
+  alias Jerboa.Params
+  alias Fennec.TURN
+
+  @lifetime 5 * 60 # MUST be 5mins
+
+  @spec service(Params.t, Fennec.client_info, Fennec.UDP.server_opts, TURN.t)
+    :: {Params.t, TURN.t}
+  def service(params, _client, _server, turn_state) do
+    request_status =
+      {:continue, params, %{}}
+      |> maybe(&verify_allocation/3, [turn_state])
+      |> maybe(&verify_xor_peer_address/2, [])
+      |> maybe(&create_permissions/3, [turn_state])
+
+    case request_status do
+      {:error, error_code} ->
+        {%{params | attributes: [error_code]}, turn_state}
+      {:respond, {new_params, new_turn_state}} ->
+        {new_params, new_turn_state}
+    end
+  end
+
+  defp verify_allocation(params, state, turn_state) do
+    case turn_state do
+      %TURN{allocation: %TURN.Allocation{}} ->
+        {:continue, params, state}
+      _ ->
+        {:error, %Attribute.ErrorCode{code: 437}}
```

Comment:
Did you consider assigning error names instead of error codes? I know it's a bit late, but now that are so many errors, I find  it hart to understand. You can do `%ErrorCode{name: :allocation_mismatch}`